### ISSUE TITLE
loss is negative and cannot overfit on mini-set of nuScenes

### ISSUE THREAD

**Issue by XinchaoGou** [wayveai/fiery#2]
Thank you for your great work and sharing the code! 
I'm trying to train a model to overfit on the mini-set for the first step. But it cannot overfit on the mini-set even after 1000 epochs. 

According to the code, you have two losses for each task, one is "factor*loss_fn" and the other is for "uncertainty". I'm confused about the part "uncertainty" and I think it may hinder the training.
![image](https://user-images.githubusercontent.com/35408641/122321961-457f3100-cf57-11eb-8edf-ab75d960c408.png)

For example, loss for instance_offset is about 0.2-0.6.

![image](https://user-images.githubusercontent.com/35408641/122324951-49618200-cf5c-11eb-88ff-0813cdaa4a9e.png)

But loss for instance_uncertainty is negative and contribute more (about -1.4 at 14k) to the final loss. So the model may just blindly decrease "uncertainty". 

![image](https://user-images.githubusercontent.com/35408641/122325045-731aa900-cf5c-11eb-9e72-2bc9222a7d51.png)

I wonder why you design the loss this why. I'll appreciate it If you have any reference or document to help me understand this part.

Best Regards,
Xinchao

**Comment by anthonyhu** [comment]
Hi Xinchao,

The uncertainty loss corresponds to the Multitask uncertainty to automatically weight the difference losses instead of tuning them, as described in "[Multi-Task Learning Using Uncertainty to Weigh Losses for Scene Geometry and Semantics, Kendall et al.](https://arxiv.org/abs/1705.07115)".

Best,
Anthony
